### PR TITLE
New feature | Code generator for interfaces and method declarations

### PR DESCRIPTION
**Pull request by DaanBiesterbos**
Hi,

I developed a generator for interfaces and the method declarations. I felt like sharing today. Reflection classes and tests are included.

Cheers,
Daan

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,7 +7,7 @@
 .*.un~
 nbproject
 tmp/
-
+composer.phar
```
Comment (Maks3w) [inline .gitignore]: please undo this. Instead add your custom files in your global gitignore

**Comment by DaanBiesterbos** [comment]
@Maks3w  Done :-)

File: src/Generator/InterfaceGenerator.php [hunk 2]
```diff
@@ -0,0 +1,733 @@
...
+
+namespace Zend\Code\Generator;
+
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @author    Daan Biesterbos <daanbiesterbos@gmail.com>
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
```
Comment (samsonasik) [inline src/Generator/InterfaceGenerator.php]: license should be on top of namespace like others. @author should be before class

**Comment by Martin-P** [comment]
Please fix all docblocks and remove stuff like this:
```
* Class InterfaceGenerator
* @package   Zend\Code\Generator
* @author    Daan Biesterbos <daanbiesterbos@gmail.com>
* @filesource
```

File: src/Generator/InterfaceGenerator.php [hunk 3]
```diff
@@ -0,0 +1,738 @@
...
+
+    /**
+     * @var bool
+     */
+    protected $flags = 0x00;
+
+    /**
+     * @var array Array of string names
+     */
+    protected $extendedInterfaces = array();
```
Comment (samsonasik) [inline src/Generator/InterfaceGenerator.php]: use short array syntax []

Comment (DaanBiesterbos) [inline src/Generator/InterfaceGenerator.php]: @Samsonik  I switched to use short array syntax. I would like to know if this is a real guideline. I see -alot- of old fashioned array() statements in this project. I don't mind using one or the other. But I don't like doing the same thing twice. Is there some sort of document available?

Comment (Maks3w) [inline src/Generator/InterfaceGenerator.php]: @DaanBiesterbos Recently the rule has been applied https://github.com/zendframework/zend-code/commit/b1ddef7792190a6f8f07d2d5607a722bfc00e8dc

**Comment by DaanBiesterbos** [comment]
Hmm.. Martin-P suggested that I had to remove the docblocks...

**Pull request closed without merging**